root/static/scripts/edit/URLCleanup.js

```diff
@@ -2540,6 +2539,19 @@ const CLEANUPS: CleanupEntries = {
       };
     },
   },
+  'eonkyo': {
+    match: [new RegExp('^(https?://)?([^/]+\\.)?e-onkyo\\.com', 'i')],
+    restrict: [LINK_TYPES.downloadpurchase],
+    clean: function (url) {
+      return url.replace(/^(?:https?:\/\/)?(?:www\.)?e-onkyo\.com\/music\/album\/([a-z]+\d+).*$/, 'https://www.e-onkyo.com/music/album/$1/');
+    },
+    validate: function (url) {
+      return {
+        result: /^https:\/\/www\.e-onkyo\.com\/music\/album\//.test(url),
```

Comment:
The URL can be fully checked to the end by reusing parts of your function `clean` above:

```suggestion
        result: /^https:\/\/www\.e-onkyo\.com\/music\/album\/[a-z]+\d+\/$/.test(url),
```